cpp/cmake/thirdparty/get_hnswlib.cmake

```diff
@@ -26,6 +26,11 @@ function(find_and_configure_hnswlib)
                 COMMAND git clone --branch=v0.6.2 https://github.com/nmslib/hnswlib.git hnswlib-src
                 WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/_deps )
 
+        message("SOURCE ${CMAKE_CURRENT_SOURCE_DIR}")
+        execute_process (
+                COMMAND git apply ${CMAKE_CURRENT_SOURCE_DIR}/cmake/patches/hnswlib.patch
```

Comment:
So how does this work w/ the original hnswlib benchmarks? We don't want to modify hnswlib for those, right? Should this patch only get applied for cagra+hnswlib? Maybe we need to check the code out into a different working directory for each and only apply the patch to one?

Reply:
We introduce a parameter in hnswlib called `base_layer_only`. When `false`, the original hnswlib is unchanged and that's what hnswlib benchmarks use

Reply:
That's fine so long as it's not modifying the basic behavior.